erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -1172,3 +1172,15 @@ def update_item(source_doc, target_doc, source_parent):
 	}, target_doc, set_missing_values)
 
 	return doclist
+
+
+@frappe.whitelist()
+def init_print_check(start, selected_docs, doctype):
+	series = int(start)
+	docs = json.loads(selected_docs)
+	for doc in docs:
+		frappe.db.set_value(doctype, doc.get("name"), {
+			"reference_no": series,
+			"reference_date": nowdate()
+		})
+		series += 1
```

Comment:
I think we should display which payment entry got assigned which number as the response, talk to @vishdha about building that message out. He's done it before for bulk-cancelling documents.